test/jdk/java/util/zip/ZipFile/EndOfCenValidation.java

```diff
@@ -0,0 +1,231 @@
+    public static final Path BAD_CEN_OFFSET_ZIP = Path.of("bad-cen-offset.zip");
+    // Some ZipFile constants for manipulating the END header
+    private static final int ENDHDR = ZipFile.ENDHDR; // End of central directory record size
+    private static final int ENDSIZ = ZipFile.ENDSIZ; // Offset of CEN size field within ENDHDR
+    private static final int ENDOFF = ZipFile.ENDOFF; // Offset of CEN offset field within ENDHDR
+    // Maximum allowed CEN size allowed by ZipFile
+    private static int MAX_CEN_SIZE = Integer.MAX_VALUE - ENDHDR - 1;
+
+    // Expected message when CEN size does not match file size
+    private static final String INVALID_CEN_BAD_SIZE = "invalid END header (bad central directory size)";
+    // Expected message when CEN offset is too large
+    private static final String INVALID_CEN_BAD_OFFSET = "invalid END header (bad central directory offset)";
+    // Expected message when CEN size is too large
+    private static final String INVALID_CEN_SIZE_TOO_LARGE = "invalid END header (central directory size too large)";
+
+    // A valid ZIP file, used as a template
+    private byte[] zipBytes;
+
+    /**
+     * Create a valid ZIP file, used as a template
+     */
+    @BeforeTest
+    public void setup() throws IOException {
+        zipBytes = templateZip();
+    }
+
+    /**
+     * Delete big files after test, in case the file system did not support sparse files.
+     */
+    @AfterTest
+    public void cleanup() throws IOException {
+        Files.deleteIfExists(CEN_TOO_LARGE_ZIP);
+        Files.deleteIfExists(INVALID_CEN_SIZE);
+        Files.deleteIfExists(BAD_CEN_OFFSET_ZIP);
+    }
+
+    /**
+     * Validates that an end of central directory record with
+     * a CEN length exceeding {@link #MAX_CEN_SIZE} limit is rejected
+     */
+    @Test
+    public void shouldRejectTooLargeCenSize() throws IOException {
+        int size = MAX_CEN_SIZE + 1;
+
+        Path zip = zipWithModifiedEndRecord(size, true, 0, CEN_TOO_LARGE_ZIP);
+
+        ZipException ex = expectThrows(ZipException.class, () -> {
+            new ZipFile(zip.toFile());
+        });
+
+        assertEquals(ex.getMessage(), INVALID_CEN_SIZE_TOO_LARGE);
+    }
+
+    /**
+     * Validate that an end of central directory record with a
+     * CEN size which exceeds the position of the EOC record is rejected.
+     */
+    @Test
+    public void shouldRejectInvalidCenSize() throws IOException {
+
+        int size = MAX_CEN_SIZE;
+
+        Path zip = zipWithModifiedEndRecord(size, false, 0, INVALID_CEN_SIZE);
+
+        ZipException ex = expectThrows(ZipException.class, () -> {
+            new ZipFile(zip.toFile());
+        });
+
+        assertEquals(ex.getMessage(), INVALID_CEN_BAD_SIZE);
+    }
+
+    /**
+     * Validate that an end of central directory record with a CEN offset which
+     * is larger than the EOC position minus the CEN size is rejected
+     * @throws IOException
```

Comment:
I don't think `@throws` javadoc comments in tests are useful; especially checked ones

Reply:
I recently added a `@throws` by review comment from Lance. I tend to agree that is not very useful, so I'm removing both ocurrences as per your recommendation.

If Lance wants it back, he can add a review comment saying so. I don't particularly care which professor gets the last word in this matter :-)

Reply:
Please add them back before pushing with `@throws IOException if an error occurs`

Yes I understand they do not add much value but it will reduce noise from IDEs and if/when we add more checks for missing javadoc tags, it will save us from  having to revisit this

Reply:
Thanks, I've added `@throws IOException if an error occurs` to all methods throwing `IOException`.

For the record, let me state my personal (rather strong) opinion:

I think this a pretty crazy level of boilerplate noise for a test and that it affects readability negatively. My IDE (IntelliJ) does not complain about this at all. Any IDE which complains should be fixed or configured to not complain. Changing Javadoc to complain about this in a scope like this  would be a poor decision.

I think the PR is ready to be sponsored after this.

Reply:
> I think the PR is ready to be sponsored after this.

Please see my comment regarding the  end of central directory record.

I would prefer to tweak that in a fashion similar to what I indicated as I thought your original version was clearer.  I understand what Martin was indicating of the use of EOC and that could have been addressed by adding  _(EOC)_ or _ENDHDR_ after _end of central directory record_ to make it clearer

I will leave it up to you as to whether you want to make the change but it would be clearer as I think we took a small step backwards.

Either way you should not need a sponsor and should be good to integrated when you are ready

Reply:
> I hear you but the option is not to use javadoc comments and use block comments :-)

I learned something new today: Javadoc comments and block comments are not the same!

Reply:
So I changed the references to this ultimate ZIP structure to read like this:

'End of central directory record' (END header)

Example:

```
/**
     * Validate that an 'End of central directory record' (END header)
     * where the value of the CEN size field exceeds the position of
     * the END header is rejected.
```

This way we stick to the 'official' ZIP verbiage but also allow using the shorter 'END header'.

What do you think?

Reply:
> Either way you should not need a sponsor and should be good to integrated when you are ready

As a non-committer, I still think I need help from a committer to `/sponsor` this change, right?

Reply:
> I think this a pretty crazy level of boilerplate noise for a test and that it affects readability negatively. 

I'm with Eirik here.

I've maintained jdk tests for 20 years and have never had to worry about nagging to add javadoc boilerplate.

doclint should restrict itself to checking what's there, not to check for completeness, in tests and private APIs.

And it looks like the jdk makefiles are in fact set up to work this way.  

I see:
./make/Docs.gmk:102:JAVADOC_DISABLED_DOCLINT_WARNINGS := missing

Especially for test methods, the contract of a test is very simple: "do nothing", but possibly throw an informative exception.  If the compiler requires a throws declaration, then I like `throws Throwable`  which evokes this contract.  A javadoc `@throws` declaration is the sort of useless boilerplate that makes people hate Java.